Components/IconTypes/IconType_customvalue/customvalue.lua

```diff
@@ -0,0 +1,130 @@
+﻿local TMW = TMW
+if not TMW then return end
+
+local L = TMW.L
+
+local print = TMW.print
+
+
+local Type = TMW.Classes.IconType:New("customvalue")
+
+Type.name = L["ICONMENU_CUSTOMVALUE"]
+Type.desc = L["ICONMENU_CUSTOMVALUE_DESC"]
+Type.menuIcon = "Interface/Icons/inv_misc_punchcards_white"
+
+Type.hasNoGCD = true
+Type.barIsTimer = false
+
+local STATE_SUCCEED = TMW.CONST.STATE.DEFAULT_SHOW
+local STATE_FAIL = TMW.CONST.STATE.DEFAULT_HIDE
+
+Type:SetAllowanceForView("icon", false)
+Type:SetModuleAllowance("IconModule_TimerBar_Overlay", false)
+Type:SetModuleAllowance("IconModule_CooldownSweep", false)
+Type:UsesAttributes("texture")
+
+Type:UsesAttributes("value")
+Type:UsesAttributes("maxValue")
+Type:UsesAttributes("valueColor")
+Type:UsesAttributes("state")
+
+local BarColors = {{r=0, g=0, b=1, a=1}, {r=0, g=1, b=1, a=1}, {r=0, g=1, b=0, a=1}}
+
+Type:RegisterIconDefaults{
+	-- Lua code to be evaluated
+	LuaCode			="",
+
+	-- Currently displayed value
+	value			=0,
+	
+	-- Maximum value displayed
+	maxValue		=0,
+	
+	-- Bar color scheme
+	valueColor		=BarColors,
+	
+	-- Initial state of icon (set as fail since we have no code yet)
+	state			=STATE_FAIL,
+
```

Comment:
None of these are settings and so do not need to be declared as defaults.

`LuaCode` here is the only setting.